docs/01-programs/02-candy-machine/02-candy-machine-settings.md

```diff
@@ -1,12 +1,289 @@
+- `address`: The address of the mint account of the Collection NFT.
+- `updateAuthority`: The update authority of the Collection NFT as a signer.
+
+Here’s an example.
+
+```tsx
+import { Keypair } from "@solana/web3.js";
+
+// Create the Collection NFT.
+const collectionAuthority = Keypair.generate();
+const { nft: collectionNft } = await metaplex.nfts().create({
+  name: "My Collection NFT",
+  uri: "https://example.com/path/to/some/json/metadata.json",
+  sellerFeeBasisPoints: 0,
+  isCollection: true,
+  updateAuthority: collectionAuthority,
+});
+
+// Pass the Collection NFT and its authority in the settings.
+const candyMachineSettings = {
+  collection: {
+    address: collectionNft.address,
+    updateAuthority: collectionAuthority,
+  },
+};
+```
+
+</div>
+</AccordionItem>
+</Accordion>
+
+## Item Settings
+
+Candy Machine settings also contain information regarding the items that are or will be loaded inside it. The **Items Available** attribute falls in that category and stores the maximum amount of NFTs that will be minted from the Candy Machine.
+
+<Accordion>
+<AccordionItem title="JS SDK" open={true}>
+<div className="accordion-item-padding">
+
+When creating a new Candy Machine, the `itemsAvailable` attribute is required and must be passed like so.
+```tsx
+import { toBigNumber } from "@metaplex-foundation/js";
+
+const candyMachineSettings = {
+  itemsAvailable: toBigNumber(500),
+};
+```
+
+</div>
+</AccordionItem>
+</Accordion>
+
+On top of the **Items Available** attribute, two other attributes define how items are loaded in the Candy Machine. You must choose exactly one of these attributes and leave the other one empty. These attributes are:
+
+- The **Config Line Settings**.
+- The **Hidden Settings**.
+
+Note that once a Candy Machine is created using one of these two modes, it cannot be updated to use the other mode. Additionally, when **Config Line Settings** are used, it is no longer possible to update the **Items Available** attribute.
+
+Let’s go through both of them in a bit more detail.
+
+### Config Line Settings
+
+The **Config Line Settings** attribute allows us to describe the items that are or will be inserted inside our Candy Machine. It enables us to keep the size of the Candy Machine to a minimum by providing exact lengths for the **Names** and **URIs** of our items as well as providing some shared prefixes to reduce that length. The **Config Line Settings** attribute is an object containing the following properties:
+
+- **Name Prefix**: A name prefix shared by all inserted items.
+- **Name Length**: The maximum length for the name of each inserted item excluding the name prefix.
+- **URI Prefix**: A URI prefix shared by all inserted items.
+- **URI Length**: The maximum length for the URI of each inserted item excluding the URI prefix.
+- **Is Sequential**: Indicates whether to mint NFTs sequentially — `true` — or in random order — `false`.
```

Comment:
Should we add a recommendation that this should be set to `true` in most cases. Is there a default value on the SDK?

Reply:
Good shout. Yes, the SDK default that to `false`. I'll document that here.